File: webfront/web/js/models/product.js
var Product = BasicModel.extend({
    modelName: 'product',
    urlRoot: baseApiUrl + "/products",

    defaults: {
        id: null,
        name: null,
        units: null,
        vat: null,
        purchasePrice: null,
        barcode: null,
        sku: null,
        vendorCountry: null,
        vendor: null,
        info: null,
        amount: null
    },

    toJSON: function(options) {
        _.defaults(options || (options = {}), {
            toSave: false
        });

        var data = BasicModel.prototype.toJSON.call(this, options)

        if(options.toSave){
            data[this.modelName].amount = undefined;
        }

        return data;
    },

    unitsEnum: {
        kg: {
            textEdit: "Килограммы",
            textView: "килограмм",
            textViewShort: "кг"
        },
        unit: {
            textEdit: "Штуки",
            textView: "штука",
            textViewShort: "шт."
        },
        liter: {
            textEdit: "Литры",
            textView: "литр",
            textViewShort: "л"
        }
    },

    vatEnum: [0, 10, 18]
});

File: autotests/src/main/java/project/lighthouse/autotests/objects/Product.java
package project.lighthouse.autotests.objects;

import org.json.JSONException;
import org.json.JSONObject;

public class Product extends AbstractObject {

    private static final String API_URL = "/products";

    public Product(JSONObject jsonObject) {
        this.jsonObject = jsonObject;
    }

    public Product(String name, String units, String vat, String purchasePrice, String barcode,
                   String sku, String vendorCountry, String vendor, String info,
                   String subCategory, String retailMarkupMax,
                   String retailMarkupMin, String rounding) throws JSONException {
        this(new JSONObject()
                .put("name", name)
                .put("units", units)
                .put("vat", vat)
                .put("purchasePrice", purchasePrice)
                .put("barcode", barcode)
                .put("sku", sku)
                .put("vendorCountry", vendorCountry)
                .put("vendor", vendor)
                .put("info", info)
                .put("subCategory", subCategory)
                .put("retailMarkupMax", retailMarkupMax)
                .put("retailMarkupMin", retailMarkupMin)
                .put("rounding", rounding)
        );
    }

    @Override
    public String getApiUrl() {
        return API_URL;
    }

    public String getSku() throws JSONException {
        return getPropertyAsString("sku");
    }
}
